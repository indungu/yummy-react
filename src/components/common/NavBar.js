import React from 'react';

const NavBar = (props) => {
  return (
    <nav className="navbar navbar-expand-lg navbar-dark bg-dark">
      <a className="navbar-brand App-title" href="/">Yummy Recipes</a>
      <button className="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
        <span className="navbar-toggler-icon"></span>
      </button>
      <div className="collapse navbar-collapse" id="navbarSupportedContent">
        <ul className="navbar-nav mr-auto">
          <li className="nav-item active">
            <a href="/">Home</a>
          </li>
        </ul>
        { !props.isLoggedIn ?
          <ul className="navbar-nav flex-row ml-md-auto d-none d-md-flex">
            <li className="nav-item active">
              <a href="/signup">Sign Up</a>
            </li>
            <li className="nav-item active">
              <a href="/login">Login</a>
            </li>
          </ul>
          :
          <ul className="navbar-nav flex-row ml-md-4 d-none d-md-flex">
            <span className="App-title text-white">Welcome {props.user}</span>
            <li className="nav-item">
              <button
                className="nav-link btn btn-outline-danger"
                style={{ color: 'white' }}
                onClick={props.onLogoutClick}
              >Logout
              </button>
            </li>
          </ul>
        }

      </div>
    </nav>
  );
};

export default NavBar;
