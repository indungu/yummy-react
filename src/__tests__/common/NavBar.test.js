import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import NavBar from '../../components/common/NavBar';

describe('Login', () => {
  const wrapper1 = shallow(<NavBar />);
  it('Renders properly', () => {
    expect(shallowToJson(wrapper1)).toMatchSnapshot();
  });
  // eslint-disable-next-line
  const wrapper2 = shallow(<NavBar isLoggedIn={true} user="Mark" />);
  it('Renders properly when user is logged in', () => {
    expect(shallowToJson(wrapper2)).toMatchSnapshot();
  });
});
