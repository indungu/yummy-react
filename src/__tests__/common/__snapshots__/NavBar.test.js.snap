// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Renders properly 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-dark bg-dark"
>
  <a
    className="navbar-brand App-title"
    href="/"
  >
    Yummy Recipes
  </a>
  <button
    aria-controls="navbarSupportedContent"
    aria-expanded="false"
    aria-label="Toggle navigation"
    className="navbar-toggler"
    data-target="#navbarSupportedContent"
    data-toggle="collapse"
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    className="collapse navbar-collapse"
    id="navbarSupportedContent"
  >
    <ul
      className="navbar-nav mr-auto"
    >
      <li
        className="nav-item active"
      >
        <a
          href="/"
        >
          Home
        </a>
      </li>
    </ul>
    <ul
      className="navbar-nav flex-row ml-md-auto d-none d-md-flex"
    >
      <li
        className="nav-item active"
      >
        <a
          href="/signup"
        >
          Sign Up
        </a>
      </li>
      <li
        className="nav-item active"
      >
        <a
          href="/login"
        >
          Login
        </a>
      </li>
    </ul>
  </div>
</nav>
`;

exports[`Login Renders properly when user is logged in 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-dark bg-dark"
>
  <a
    className="navbar-brand App-title"
    href="/"
  >
    Yummy Recipes
  </a>
  <button
    aria-controls="navbarSupportedContent"
    aria-expanded="false"
    aria-label="Toggle navigation"
    className="navbar-toggler"
    data-target="#navbarSupportedContent"
    data-toggle="collapse"
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    className="collapse navbar-collapse"
    id="navbarSupportedContent"
  >
    <ul
      className="navbar-nav mr-auto"
    >
      <li
        className="nav-item active"
      >
        <a
          href="/"
        >
          Home
        </a>
      </li>
    </ul>
    <ul
      className="navbar-nav flex-row ml-md-4 d-none d-md-flex"
    >
      <span
        className="App-title text-white"
      >
        Welcome 
        Mark
      </span>
      <li
        className="nav-item"
      >
        <button
          className="nav-link btn btn-outline-danger"
          style={
            Object {
              "color": "white",
            }
          }
        >
          Logout
        </button>
      </li>
    </ul>
  </div>
</nav>
`;
